web/src/main/java/org/mskcc/cbio/oncokb/api/pub/v1/AnnotationsApiController.java

```diff
@@ -517,7 +519,186 @@ public ResponseEntity<List<IndicatorQueryResp>> annotateStructuralVariantsPost(
+            annotationSearchResult.setQueryType(AnnotationSearchQueryType.GENE);
+            annotationSearchResult.setIndicatorQueryResp(IndicatorUtils.processQuery(query, null, null, null));
+            result.add(annotationSearchResult);
+        }
+        return result;
+    }
+
+    private LinkedHashSet<AnnotationSearchResult> findActionableGenesByAlterationSearch(String keyword) {
+        LinkedHashSet<AnnotationSearchResult> result = new LinkedHashSet<>();
+        List<Alteration> altMatches = AlterationUtils.lookupVariant(keyword, false, true, AlterationUtils.getAllAlterations());
+        for (Alteration alteration: altMatches) {
+            Query indicatorQuery = new Query();
+            if (alteration.getName().toLowerCase().contains(keyword.toLowerCase())) {
+                indicatorQuery.setAlteration(alteration.getName());
+            } else {
+                indicatorQuery.setAlteration((alteration.getAlteration()));
+            }
+            indicatorQuery.setEntrezGeneId(alteration.getGene().getEntrezGeneId());
+            indicatorQuery.setHugoSymbol(alteration.getGene().getHugoSymbol());
+            AnnotationSearchResult annotationSearchResult = new AnnotationSearchResult();
+            annotationSearchResult.setQueryType(AnnotationSearchQueryType.VARIANT);
+            annotationSearchResult.setIndicatorQueryResp(IndicatorUtils.processQuery(indicatorQuery, null, null, null));
+            result.add(annotationSearchResult);
+        }
+        return result;
+    }
+
+    private LinkedHashSet<AnnotationSearchResult> findActionableGenesByCancerType(String query) {
 
+        Set<Evidence> allImplicationEvidences = EvidenceUtils.getEvidenceByEvidenceTypesAndLevels(EvidenceTypeUtils.getImplicationEvidenceTypes(), LevelUtils.getPublicLevels());
+
+        query = query.toLowerCase();
+
+        Set<TumorType> tumorTypeMatches = new HashSet<>();
+        for(Map.Entry<String, TumorType> subtype: CacheUtils.getLowercaseSubtypeTumorTypeMap().entrySet()) {
+            if(subtype.getKey().contains(query)) {
+                tumorTypeMatches.add(subtype.getValue());
+            }
+        }
+
+        for(Map.Entry<String, TumorType> mainType: CacheUtils.getMainTypeTumorTypeMap().entrySet()) {
+            if (mainType.getKey().toLowerCase().contains(query)) {
+                tumorTypeMatches.add(mainType.getValue());
+            }
+        }
+
+        if (tumorTypeMatches.isEmpty()) {
+            return new LinkedHashSet<>();
+        }
+
+        LinkedHashSet<AnnotationSearchResult> result = new LinkedHashSet<>();
+        Set<SearchObject> searchObjects = new HashSet<>();
+        for (TumorType tumorType : tumorTypeMatches) {
+            for (Evidence evidence : allImplicationEvidences) {
+                if (evidence.getGene().getHugoSymbol().equals("NTRK3")) {
+                    Boolean t = true;
+                }
```

Comment:
What's this?

Reply:
Thanks for the catch. I was using it to debug some scenarios. Removed.